semgrep-core/src/experiments/datalog/Datalog_experiment.ml

```diff
@@ -70,25 +70,8 @@ let dump_il file =
   let lang = List.hd (Lang.langs_of_filename file) in
   let ast = Parse_target.parse_program file in
   Naming_AST.resolve lang ast;
-
-  let v =
-    V.mk_visitor
-      {
-        V.default_visitor with
-        V.kfunction_definition =
-          (fun (_k, _) def ->
-            let s =
-              AST_generic.show_any (G.S (H.funcbody_to_stmt def.G.fbody))
-            in
-            pr2 s;
-            pr2 "==>";
-
-            let _, xs = AST_to_IL.function_definition lang def in
-            let s = IL.show_any (IL.Ss xs) in
-            pr2 s);
-      }
-  in
-  v (G.Pr ast)
+  let xs = AST_to_IL.stmt lang (G.stmt1 ast) in
+  List.iter (fun stmt -> pr2 (IL.show_stmt stmt)) xs
```

Comment:
Doesn't this require #5787 to be merged first to work as you describe it?

Reply:
No, this is just a general change to print out the top-level statements in the IL. This is something that isn't done either way.

#5787 just makes new, shinier nodes at the top-level of the IL.

Reply:
Without #5787, `AST_to_IL.stmt` is not going to translate `DefStmt`s with function definitions to IL, correct? So when you `IL.show_stmt` those statements, it will dump the Generic rather than the IL.

Reply:
Without #5787, `AST_to_IL.stmt` still translates `DefStmt`s which are function definitions, it's just that it's translated to a more "degenerate" form. cf. 
<img width="526" alt="image" src="https://user-images.githubusercontent.com/49291449/181588854-99512914-f7ef-4214-bfb0-e5cbc8d73081.png">

I suppose what you mean by this is that the new `MiscStmt` is not really a "translation", since it just keeps the generic AST representation around. It's true that this will not show the translated version, since they're kept the same, but I think that that can only be a good thing, as it's representative of the true form of the data, which is just an encapsulated `DefStmt`.

Reply:
It's fine with me to have the program-level dump that you propose, but we still need to be able to dump the IL of the individual functions in the program in some other way. Because if taint doesn't work one of the first things I do is to check the IL translation of the function(s) involved.